apis/v1alpha1/bucket_types.go

```diff
@@ -46,6 +46,7 @@ type BucketParameters struct {
 	RP *string `json:"rp,omitempty"`
 
 	// Rules to expire or retain data. No rules means data never expires.
+	// +optional
 	RetentionRules []RetentionRule `json:"retentionRules"`
```

Comment:
```suggestion
	RetentionRules []RetentionRule `json:"retentionRules,omitempty"`
```

Otherwise, it will be empty array on the wire. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
Addressed in 5e7a6bd.